Migrate from self-hosted to Teleport-enterprise cloud

oshati opened the pull request:
Documentation guide for Migrating from a self-hosted Teleport deployment to the Teleport Cloud

oshati committed:
Migrate from self-hosted to Teleport-enterprise cloud

oshati committed:
Update migrating.mdx

oshati committed:
update guide

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Jeff Anderson <jeff@goteleport.com>

oshati committed:
update cloud migration doc

webvictim commented on docs/pages/migrating.mdx:
````diff
@@ -0,0 +1,168 @@
+$ tctl get users > users.yaml 
+```
+
+- Once you have the resource configuration file from the above, proceed to log in to your Teleport Cloud tenant with an admin user and create the resources from the exported files
````
```suggestion
- Once you have the resource configuration file from the above, proceed to log in to your Teleport Cloud tenant with an admin user and create the resources from the exported files:

```

webvictim commented on docs/pages/migrating.mdx:
````diff
@@ -0,0 +1,168 @@
+
+```code
+$ tsh login --proxy=example.teleport.sh --user=admin@example.com
+$ tctl create -f  roles.yaml
+$ tctl create -f  user.yaml
````
```suggestion
$ tctl create -f roles.yaml
$ tctl create -f user.yaml

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+For your SSO auth connector, most SSO integrations only work for a single configured endpoint. It is recommended to create a separate SSO connector in your Identity Provider specifically for the Teleport Cloud endpoint, and configure a new Auth Connector in the Teleport Cloud tenant.
+
+
+### Migrate Teleport Agents.
```
```suggestion
### Migrate Teleport Agents

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+### Migrate Teleport Agents.
+
+To ensure a successful migration process across your infrastructure, start by cataloging the various services you're managing with Teleport, The following resources should be considered for migration:
```
```suggestion
To ensure a successful migration process across your infrastructure, start by cataloging the various services you're managing with Teleport. The following resources should be considered for migration:

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+### Migrate Teleport Agents.
+
+To ensure a successful migration process across your infrastructure, start by cataloging the various services you're managing with Teleport, The following resources should be considered for migration:
+Teleport Agents (SSH, Kubernetes, Databases, Applications, Desktops)
+Machine ID Bots
+Access Request Plugins
```
```suggestion
- Teleport Agents (SSH, Kubernetes, Databases, Applications, Desktops)
- Machine ID Bots
- Access Request Plugins

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+The following procedure is required to successfully migrate your Teleport agents:
+
+- Log in to your Teleport Enterpise cloud Tenant, You'll need to authenticate to access your Teleport Cloud resources and management interfaces.
```
```suggestion
- Log in to your Teleport Enterpise Cloud tenant. You'll need to authenticate to access your Teleport Cloud resources and management interfaces.

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+- Obtain a valid join token: We recommend using a [Delegated Join Method](https://goteleport.com/docs/reference/join-methods/#delegated-join-methods)
+
+If using ephemeral tokens, ensure you specify the appropriate token type to match the Teleport services. Token types can include `node`, `app`, `Kube`, `DB`, `WindowsDesktop`, and others depending on the service you wish to join to your Teleport cluster.
```
I wasn't sure whether you were mentioning the `tctl tokens add` syntax here or the delegated join `Node`, `App`, `DB` type syntax so I went with the former.

```suggestion
If using ephemeral tokens, ensure you specify the appropriate token type to match the Teleport services. Token types can include `node`, `app`, `kube`, `db`, `windowsdesktop` and others depending on the service you wish to join to your Teleport cluster.

```

webvictim commented on docs/pages/migrating.mdx:
````diff
@@ -0,0 +1,168 @@
+$ tctl tokens add --type node,app,db --ttl 15m
+```
+
+In this command, we assigned the token the `node`, `app` and `db` type, indicating that it will belong to an SSH Service, Database Service and Application Service instance.
````
```suggestion
In this command, we assigned the token the `node`, `app` and `db` type, indicating that it will allow an agent to join which is running the Teleport `ssh_service`, `db_service` and `app_service`.

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+- Stop the Teleport service on the agent (if applicable)
+
+- Update the `proxy_server` or `auth_server` field in the agent configuration file to point to the address of your Teleport cloud proxy. By default, the configuration is located in the `/etc/teleport.yaml` directory
```
```suggestion
- Update the `proxy_server` or `auth_servers` field in the agent configuration file to point to the address of your Teleport Cloud cluster. By default, the configuration is located in the `/etc/teleport.yaml` directory.

```

webvictim commented on docs/pages/migrating.mdx:
````diff
@@ -0,0 +1,168 @@
+- example.teleport.sh:443
+```
+
+- Update the auth_token field with the newly generated token. 
````
````suggestion
- Update the `auth_token` or `join_params.token_name` field with the newly generated token.

```yaml
teleport:
  join_params:
    method: token
    token_name: new-token-goes-here
```

````

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+- Update the auth_token field with the newly generated token. 
+
+- Delete the local agent cache to force the agent to rejoin the new Teleport Cloud cluster. By default, the data is located in the  `/var/lib/teleport` directory
```
````suggestion
- Delete the local agent cache to force the agent to rejoin the new Teleport Cloud cluster. By default, the data is located in the  `/var/lib/teleport` directory:

```code
rm -rf /var/lib/teleport
```

````

webvictim commented on docs/pages/migrating.mdx:
````diff
@@ -0,0 +1,168 @@
+
+- Update the `proxy_server` or `auth_server` field in the agent configuration file to point to the address of your Teleport cloud proxy. By default, the configuration is located in the `/etc/teleport.yaml` directory
+
+```code
+proxy_server:
+- example.teleport.sh:443
+```
````
We should try and teach people to move to newer config versions at the same time IMO, which means using `version: v3` and specifying `proxy_server` instead of `auth_servers`.

````suggestion
```yaml
version: v3
teleport:
  proxy_server: example.teleport.sh:443
```
````

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+- Delete the local agent cache to force the agent to rejoin the new Teleport Cloud cluster. By default, the data is located in the  `/var/lib/teleport` directory
+
+- Start the Teleport process on each agent or recycle the pod to apply the new configuration. This will cause the agents to re-register with the Teleport cloud cluster and obtain new certificates signed by the cloud cluster's certificate authority.
```
```suggestion
- Start the Teleport process on each agent or recycle the pod to apply the new configuration. This will cause the agents to re-register with the Teleport Cloud cluster and obtain new certificates signed by the Teleport Cloud cluster's certificate authority.

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+As new services come online, they start sending heartbeat requests every few seconds to the Auth Service. This allows users to explore cluster membership. 
+
+Note: ​​Migration of agents can be done all at once or gradually, depending on your business requirements. If running teleport at scale, you'll generally want to leverage the use of a configuration management tool to automate and streamline the process of carrying out the actions involved in migrating agent configurations. 
```
```suggestion
Note: ​​Migration of agents can be done all at once or gradually, depending on your business requirements. If running Teleport at scale, you'll generally want to leverage the use of a configuration management tool to automate and streamline the process of carrying out the actions involved in migrating agent configurations. 

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+Note: ​​Migration of agents can be done all at once or gradually, depending on your business requirements. If running teleport at scale, you'll generally want to leverage the use of a configuration management tool to automate and streamline the process of carrying out the actions involved in migrating agent configurations. 
+
+
+### Verify end User Access and performance.
```
```suggestion
### Verify end User Access and performance

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+### Verify end User Access and performance.
+
+Once the migration is complete, validate that the resources are present in the Teleport cluster and verify their status and connectivity to ensure they are properly registered and available for connections. You can either check via the Web UI or use tctl to get a list of all resources and verify their registration and status. 
```
```suggestion
Once the migration is complete, validate that the resources are present in the Teleport cluster and verify their status and connectivity to ensure they are properly registered and available for connections. You can either check via the Web UI, or use `tctl` to get a list of all resources and verify their registration and status. 

```

webvictim commented on docs/pages/migrating.mdx:
````diff
@@ -0,0 +1,168 @@
+Similarly, to list all registered Kubernetes clusters, you can run the command below:
+
+```
+tctl get kube_services
````
`tctl kube ls` is probably a better bet as it will show the name, version and labels of each in a more human-readable form.

```suggestion
tctl kube ls
```

webvictim replied:
Can we also add sections for `tctl db ls`, `tctl apps ls` and `tctl get windows_desktop` here so people have the full range in one place?

webvictim commented on docs/pages/migrating.mdx:
````diff
@@ -0,0 +1,168 @@
+tctl get kube_services
+```
+
+These tctl commands will show you the resources registered within your Teleport cluster, and you can verify their availability for connections
````
```suggestion
These `tctl` commands will show you the resources registered within your Teleport cluster, and you can verify their availability for connections.

```

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+- Ensure that end users have the expected SSO access to your infrastructure and workflow.
+
+Additionally, establish break glass access procedures to ensure access to infrastructure in case of Teleport Cloud unavailability.
```
(how should people do this?!)

webvictim commented on docs/pages/migrating.mdx:
```diff
@@ -0,0 +1,168 @@
+
+### Conclusion
+
+Migrating from a self-hosted Teleport deployment to Teleport Cloud offers scalability, reliability, and ease of management. By following the steps outlined in this guide, you can transition your Teleport self-hosted deployment to the cloud successfully while ensuring minimal disruption to your operations.
```
```suggestion
Migrating from a self-hosted Teleport deployment to Teleport Cloud offers scalability, reliability, and ease of management. By following the steps outlined in this guide, you can transition your Teleport self-hosted deployment to Teleport Cloud successfully while ensuring minimal disruption to your operations.

```

webvictim reviewed (commented):
Great work! There's lots of little nits here and it seems like a lot, but I promise I'm saving you time when this goes to docs review!

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

webvictim reviewed (commented):
LGTM with a few more nits and pending Vercel preview

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/migrating.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
update cloud migration doc

oshati committed:
update cloud migration doc

ptgott commented:
Suggested edits and organization changes here: https://github.com/gravitational/teleport/pull/41285

ptgott committed:
Edit the Enterprise migration guide (#41285)

* Edit the Enterprise migration guide

**Edit the guide for structure and style:**
   - Use title case in the title.
   - Wrap lines at 80 chars.
   - Begin H2s with "Step n/d"
   - Move the Conclusion text to the intro. How-to guides do not include
     conclusions in our docs, and this information is valuable at the start
     of the guide.
   - Use numbered steps for instructions.
   - Use relative file paths for links.
   - Minor style edits.

**Remove the "Verify connectivity" step:** This repeats instructions
from the step to deploy a cloud-hosted cluster.

**Add a note on migrating plugins and Machine ID Bots:** the guide only
includes instructions for migrating agents in the section on migrating
services. This change suggests linking to docs on migrating plugins and
Bots.

**Edit the navigation:**
  - Add a link to the navigation sidebar.
  - Edit the "Choose a Teleport Edition" intro page to (a) imply that
    cloud-hosted is the recommended solution and (b) link to the
    migration guide.

* Respond to webvictim feedback

- Reduce the overuse of `Teleport Enterprise (cloud|self-hosted)`.
- Move the note re: trusted clusters to the Prerequisites.
- Mention the `--auth` flag in `tsh login` commands.
- Fix typo.
- Add a note about the `proxy_server` field

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,341 @@
+h1: Migrate to Cloud-Hosted Teleport Enterprise 
+---
+
+Migrating from a self-hosted Teleport Enterprise  deployment to a cloud-hosted
```
```suggestion
Migrating from a self-hosted Teleport Enterprise deployment to a cloud-hosted

```

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,341 @@
+
+## How it works
+
+While in a cloud-hosted Enterprise Teleport account, Teleport manages the Auth
```
```suggestion
While in a cloud-hosted Teleport Enterprise account, Teleport manages the Auth

```

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,341 @@
+## How it works
+
+While in a cloud-hosted Enterprise Teleport account, Teleport manages the Auth
+Service and Proxy Service for you, you need to migrate dynamic resources and
```
```suggestion
Service and Proxy Service for you, but you need to migrate dynamic resources and

```

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,341 @@
+
+1. Set up a separate cloud-hosted Teleport account.
+1. Retrieve dynamic Teleport resources from the Auth Service backend on the
+   self-hosted cluster and apply them against the Auth Service backed on the
```
```suggestion
   self-hosted cluster and apply them against the Auth Service backend on the

```

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,341 @@
+- `dbs`
+- `login_rule`
+
+If you are using infrastructure-as-code tools to 
+To achieve this:
```
This looks like a fragment which isn't going anywhere; the instructions below aren't related to IaC.

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
````diff
@@ -0,0 +1,341 @@
+   the new cloud-hosted Teleport Enterprise cluster. By default, the data is
+   located in the  `/var/lib/teleport` directory:
+
+   ```code
+   rm -rf /var/lib/teleport
+   ```
````
We're also missing instructions here about how to wipe out existing state to rejoin a Kubernetes agent to a new cluster. Essentially it's something like:

```
# get the release name
helm -n <namespace> ls
# delete the state secret
kubectl -n <namespace> delete secret <release-name>-0-state
```

oshati committed:
Update docs/pages/choose-an-edition/migrate-to-cloud.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/choose-an-edition/migrate-to-cloud.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/choose-an-edition/migrate-to-cloud.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/choose-an-edition/migrate-to-cloud.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
Update docs/pages/choose-an-edition/migrate-to-cloud.mdx

Co-authored-by: Gus Luxton <gus@goteleport.com>

oshati committed:
fix docs

oshati committed:
fix docs

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,352 @@
+h1: Migrate to Cloud-Hosted Teleport Enterprise 
+---
+
+Migrating from a self-hosted Teleport Enterprise deployment to a cloud-hosted
+
```
```suggestion
```
This isn't a full sentence, it looks like it can just be removed.

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,352 @@
+## How it works
+
+While in a cloud-hosted Teleport Enterprise account, Teleport manages the Auth
+
```
```suggestion
```
Remove extra line break

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,352 @@
+While in a cloud-hosted Teleport Enterprise account, Teleport manages the Auth
+
+Service and Proxy Service for you, but you need to migrate dynamic resources and
+
```
```suggestion
```
Remove extra line break

webvictim commented on docs/pages/choose-an-edition/migrate-to-cloud.mdx:
```diff
@@ -0,0 +1,352 @@
+1. Set up a separate cloud-hosted Teleport account.
+1. Retrieve dynamic Teleport resources from the Auth Service backend on the
+   self-hosted cluster and apply them against the Auth Service backend on the
+
```
```suggestion
```
Remove extra line break

ptgott committed:
Apply suggestions from code review

Co-authored-by: Gus Luxton <gus@goteleport.com>

ptgott committed:
Fix linter issues
